cmd/bucket-replication.go

```diff
@@ -683,9 +672,12 @@ func initBackgroundReplication(ctx context.Context, objectAPI ObjectLayer) {
 	if globalReplicationState == nil {
 		return
 	}
-
-	// Start with globalReplicationConcurrent.
-	for i := 0; i < globalReplicationConcurrent; i++ {
+	numConcurrent := globalAPIConfig.getReplicationWorkers()
+	if numConcurrent == 0 {
+		numConcurrent = 1
+	}
+	// Start with numConcurrent workers.
+	for i := 0; i < numConcurrent; i++ {
```

Comment:
```suggestion
	numConcurrent := globalAPIConfig.getReplicationWorkers()
	if numConcurrent == 0 {
		numConcurrent = 1
	}
	// Start with numConcurrent workers.
	for i := 0; i < numConcurrent; i++ {
```

don't need this code - you can assume that getReplicationWorkers() is always non zero.